lib/graphql/batch/loaded_field_extension.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+if Gem::Version.new(GraphQL::VERSION) < Gem::Version.new("1.9.0")
+  raise "GraphQL::Batch::LoadedFieldExtension is not supported on graphql gem versions less than 1.9"
+end
+
+require_relative 'loaded_field_extension/loader'
+
+module GraphQL::Batch
+  # Resolve the field using a class method on the GraphQL::Schema::Object
+  # for multiple instances. This avoids the need to extract the logic
```

Comment:
It's clearer in the example code, but worth specifying that the `instance` in all these cases is the schema object and not the backing implementation object.